### PR TITLE
docs/plugin/framework: Clarify new capabilities provided by v0.3.0

### PR DESCRIPTION
**Pull request by bflad**
Closes hashicorp/terraform-plugin-framework#139

## Labels

- [ ] inaccuracy
- [x] clarification
- [ ] new docs
- [ ] cosmetic bug - fixing broken text or markup
- [ ] enhancement - changing the website's behavior/layout
- [ ] api - requires an update to the [changelog](https://www.terraform.io/docs/cloud/api/changelog.html)

### CODE REVIEW

File: content/source/docs/plugin/framework/types.html.md [hunk 1]
```diff
@@ -236,10 +236,6 @@ State, Config, and Plan](/docs/plugin/framework/accessing-values.html).
 
 You may want to build your own attribute value and type implementations to allow your provider to combine validation, description, and plan customization behaviors into a reusable bundle. This helps avoid duplication or reimplementation and ensures consistency.
 
-~> **Important:** Specifying validation and plan customization for attribute
```
Comment (paddycarver) [inline content/source/docs/plugin/framework/types.html.md:239]: I think plan customization is still not supported.

**Review by paddycarver — CHANGES_REQUESTED** [review]

File: content/source/docs/plugin/framework/types.html.md (hunk 1 above)
Comment (bflad) [reply content/source/docs/plugin/framework/types.html.md:239]: Are you sure? There is now [`tfsdk.Attribute.PlanModifiers`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework@v0.3.0/tfsdk#Attribute.PlanModifiers), unless plan customization means something something else.

Comment (paddycarver) [reply content/source/docs/plugin/framework/types.html.md:239]: The full quote is "Specifying validation and plan customization for attribute **types** is not yet supported." We [have that](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/attr#TypeWithValidate) for Validation, but not for plan modification. Though we do, as you rightly point out, have it for listing modifiers on a schema attribute, we can't attach plan modification to an `attr.Type` just yet.

Comment (bflad) [reply content/source/docs/plugin/framework/types.html.md:239]: Ah sorry, was missing the _type_ part there. 🤦  I will add this back in with just a note about plan customization on types.

**Review by paddycarver — APPROVED** [review]
LGTM, let's do it. Awesome job. 👍 🚀

**Pull request merged**